src/libraries/Common/src/System/Security/Cryptography/RSAAndroid.cs

```diff
@@ -32,7 +32,7 @@ public RSAAndroid(int keySize)
 
             internal RSAAndroid(SafeRsaHandle key)
             {
-                _key = new Lazy<SafeRsaHandle>(key);
+                _key = new Lazy<SafeRsaHandle>(key.DuplicateHandle());
```

Comment:
Are there any callers that were previously using DuplicateKey that should stop doing that now so we don't finalize an intermediate value?

Reply:
No, this was a bug from integrating the certs with RSA.